### PR TITLE
Improve CI to allow PR from forks

### PR DESCRIPTION
**Pull request by ironashram**
Signed-off-by: Michele Palazzi <michele.palazzi@sysdig.com>

## What this PR does / why we need it:

The idea is to avoid a cryptic CI failure in case a PR is opened from a fork of this repo.
Still Needs more testing

**Branch PR**
All checks will run immediately (no change)

**External Fork PR**
Helm Unit tests will run automatically since they do not require secrets
The Lint and Test Charts will be skipped, then if someone with write access to this repo adds a comment `/approve-test-run sha=CommitSHA` a `repository_dispatch` will trigger the workflow which in this case will work since it will have access to secrets and it will checkout `"refs/pull/${{ github.event.client_payload.pull_request.number }}/merge"`

This should improve our way of handling contributions from forks without security concerns

One downside is that right now it's not D.R.Y., an option to consider could be leveraging the reusable workflows https://docs.github.com/en/actions/using-workflows/reusing-workflows

Based on https://github.com/imjohnbo/ok-to-test

## Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [ ] Title of the PR starts with chart name (e.g. [mychartname])
- [ ] Chart Version bumped for the respective charts
- [ ] Changelog updated for the charts with version updates.
- [ ] Variables are documented in the README.md (or README.tpl in some charts)
- [ ] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [ ] All test files are added in the tests folder of their respective chart and have a "_test" suffix

Check Contribution guidelines in README.md for more insight.

### CODE REVIEW

**Comment by yashgiri** [comment]
The changes for the Unit tests workflow are fine with me as I just wanted to prioritize parallelization. This change is pretty important as it unblocks the maintainers of the repository from attending to issues that prop up when pushes are done from a fork by people not having write access. Although testing will need to be done to see to it that it works as expected.

Regarding DRY maybe we can create a new ticket after this to optimize the same.

**Comment by yashgiri** [comment]
I would like to have @aroberts87 review on this as well.

**Comment by yashgiri** [comment]
I haven't been able to put an approval since the reviewer request has not gone out to me. Anyway, its LGTM but if approval is needed ill give it asap whenever the request reaches me.

**Comment by aroberts87** [comment]
yeesh. i didn't realize adding myself to the reviewers would clear out folks already there. sorry for the noise.

**Review by yashgiri — APPROVED** [review]

**Comment by ironashram** [comment]
@yashgiri @aroberts87 
one thing that will be needed for this to work is to check and probably change the branch protection rules of this repo 
Otherwise we will have the `lint-test` workflow (old one) stuck in `Expected — Waiting for status to be reported`, but i don't have the permisisons required to check it.

**Review by aroberts87 — APPROVED** [review]
This looks good to me, @ironashram . Thank you for making these changes. I have witnessed several times already where opening a PR from a forked repo causes a problem. This will definitely make some developers happy. Yourself included. 🙂

**Pull request merged**
